bin/analyzeShearTest.py

```diff
@@ -0,0 +1,265 @@
+    """
+
+    base_dir         Name of the subdirectory containing run_params file
+    out_cat          Name of the catalog for output. .fits is appended
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("base_dir", type=str, help="Name of the directory containing run_params file")
+    parser.add_argument("out_cat", type=str, help="Name of the catalog for output")
+    args = parser.parse_args()
+
+    outSourceCat = getCatalog(args.out_cat + ".fits")
+    schema = outSourceCat.getSchema()
+
+    #   find all the keys we need to output the summary of each subfield
+    filterKey = schema.find("filter").getKey()
+    seeingKey = schema.find("seeing").getKey()
+    shearKey = schema.find("shear").getKey()
+    nsubKey = schema.find("nsub").getKey()
+    nsourceKey = schema.find("nsource").getKey()
+    gKey = schema.find("g").getKey()
+    eAvgKey = schema.find("eAvg").getKey()
+    eStdKey = schema.find("eStd").getKey()
+    e1DevKey = schema.find("e1Dev").getKey()
+    e1StdKey = schema.find("e1Std").getKey()
+    e2DevKey = schema.find("e2Dev").getKey()
+    e2StdKey = schema.find("e2Std").getKey()
+
+    #   open the run_params file for this run and extract what we need.
+    fin = open(os.path.join(args.base_dir, "run_params"), 'r')
+    params = dict()
+    for line in fin:
+        if line.isspace():
+            continue
+        inp = line.split()
+        if inp[0][0] == '#':
+            continue
+        params[inp[0]] = inp[1]
+
+    shear_value = float(params["shear_value"])
+    psf_file = os.path.basename(params["psf_dir"])
+    filter =  int(params["filter"])
+    seeing  = float(params["seeing"])
+    n_subfields = int(params["n_subfields"])
+    n_epochs = int(params["n_epochs"])
+    galaxy_stamp_size = int(params["galaxy_stamp_size"])
+    output_dir = params["output_dir"]
+
+    #   This tells us the algorithm fields we need to extract the ellipticity and error
+    #   The shape type can be "moments" or "ellipticity"
+    #   The error can be extracted from an ellipticity error field or based on the
+    #   signal-to-noise of some other field such as flux and flux error.  In that
+    #   case, the weighting value used is shape_field/(sigma_field/sigma_signal_field)
+    if "shape_field" in params.keys():
+        shape_field = params["shape_field"] + "_"
+        shape_type = params["shape_type"]
+        if "sigma_field" in params.keys():
+            sigma_field = params["sigma_field"]
+            if "sigma_signal_field" in params.keys():
+                sigma_signal_field = params["sigma_signal_field"]
+            else:
+                sigma_signal_field = None
+        else:
+            sigma_field = None
+            sigma_signal_field = None
+    else:
+        #   If shape_field is not in params, the ellipticity and sigma field are here:
+        shape_field = ""
+        shape_type = "ellipticity"
+        sigma_field = "e_sigma"
+        sigma_signal_field = None
```

Comment:
Seems like it'd be better to use pex_config for these config values, if you're just parsing them all yourself.